### PR TITLE
Orca pytorch superresolution

### PR DESCRIPTION
**Pull request by yaxinchen666**
Add different backends for orca pytorch super resolution example.

### CODE REVIEW

File: pyzoo/zoo/examples/orca/learn/pytorch/super_resolution/README.md [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
+We demonstrate how to easily run synchronous distributed Pytorch training using Pytorch Estimator of Project Orca in Analytics Zoo. This is an example using the efficient sub-pixel convolution layer to train on [BSDS3000 dataset](https://www2.eecs.berkeley.edu/Research/Projects/CS/vision/bsds/), using crops from the 200 training images, and evaluating on crops of the 100 test images. See [here](https://github.com/pytorch/examples/tree/master/super_resolution) for the original single-node version of this example provided by Pytorch. We provide two distributed PyTorch training backends for this example, namely "bigdl" and "torch_distributed". You can run with either backend as you wish.
 
 ## Prepare the environment
 We recommend you to use [Anaconda](https://www.anaconda.com/distribution/#linux) to prepare the environment, especially if you want to run on a yarn cluster (yarn-client mode only).
 ```
 conda create -n zoo python=3.7  # "zoo" is conda environment name, you can use any name you like.
 conda activate zoo
-pip install analytics-zoo[ray]  # 0.9.0 or above
+pip install analytics-zoo[ray]  # 0.10.0.dev3 or above
```
Comment (hkvision) [inline pyzoo/zoo/examples/orca/learn/pytorch/super_resolution/README.md:10]: same as the other two examples, for torch_distributed backend?

File: pyzoo/zoo/examples/orca/learn/pytorch/super_resolution/README.md [hunk 2]
```diff
@@ -43,11 +59,26 @@ python super_resolution.py --cluster_mode yarn
 * `--cluster_mode` The mode of spark cluster. Either "local" or "yarn". Default is "local".
```
Comment (hkvision) [inline pyzoo/zoo/examples/orca/learn/pytorch/super_resolution/README.md:59]: Add --backend here?

File: pyzoo/zoo/examples/orca/learn/pytorch/super_resolution/super_resolution.py [hunk 3]
```diff
@@ -214,50 +219,84 @@ def model_creator(config):
...
+            print("===> Epoch {} Complete: Avg. Loss: {:.4f}"
+                  .format(epoch, epochinfo["train_loss"]))
+
+    def test():
+        val_stats = estimator.evaluate(data=validation_data_creator,
+                                       batch_size=opt.test_batch_size)
+        print("===> Validation Complete: Avg. PSNR: {:.4f} dB, Avg. Loss: {:.4f}"
+              .format(10 * log10(1. / val_stats["val_loss"]), val_stats["val_loss"]))
+
+    def checkpoint(epoch):
```
Comment (hkvision) [inline pyzoo/zoo/examples/orca/learn/pytorch/super_resolution/super_resolution.py:286]: Do we still need these functions? Just directly put the contents in the for loop?

**Comment by hkvision** [comment]
Also need to add tests for bigdl backend.

**Comment by yaxinchen666** [comment]
Jenkins:
http://10.239.47.210:18888/job/ZOO-PR-Validation/5334/
ZOO-PR-Python-Spark-2.2-py36 failed. Reran and passed:
http://10.239.47.210:18888/job/ZOO-PR-Python-Spark-2.2-py36/5352/

File: pyzoo/zoo/examples/orca/learn/pytorch/super_resolution/super_resolution.py [hunk 4]
```diff
@@ -214,50 +219,85 @@ def model_creator(config):
...
+            print("===> Epoch {} Complete: Avg. Loss: {:.4f}"
+                  .format(epoch, epochinfo["train_loss"]))
+
+        val_stats = estimator.evaluate(data=validation_data_creator,
+                                       batch_size=opt.test_batch_size)
+        print("===> Validation Complete: Avg. PSNR: {:.4f} dB, Avg. Loss: {:.4f}"
+              .format(10 * log10(1. / val_stats["val_loss"]), val_stats["val_loss"]))
+
+        if not exists(model_dir):
+            makedirs(model_dir)
```
Comment (hkvision) [inline pyzoo/zoo/examples/orca/learn/pytorch/super_resolution/super_resolution.py:294]: put this out of the for loop

**Review by hkvision — APPROVED** [review]

**Comment by yaxinchen666** [comment]
Jenkins:
http://10.239.47.210:18888/job/ZOO-PR-Validation/5342/
ZOO-PR-Python-Spark-3.0-py36 and ZOO-PR-Python-ExampleTests-py37-Spark3 failed. Reran and passed:
http://10.239.47.210:18888/job/ZOO-PR-Python-Spark-3.0-py36/762/
http://10.239.47.210:18888/job/ZOO-PR-Python-ExampleTests-py37-Spark3/784/
